rustup-install 0.1.0

nick0602 opened the pull request:
![TagCheck](https://bitrise-steplib-git-check.herokuapp.com/tag?pr=3572)

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

---

Bash shell to install the **Rust Toolchain** and make it available in a Bitrise Workflow.
Example builds are available here [bitrise-step-rustup](https://app.bitrise.io/app/5c455310b4a85e12) (it's kinda barebone for now but I'll update the workflows later on).

Output of `bitrise run audit-this-step` on **0.1.0** ([1ff9372e7e36e2d60003e4237709b53df0ef2c97](https://github.com/nick0602/bitrise-step-rustup/commit/1ff9372e7e36e2d60003e4237709b53df0ef2c97)):
```shell
➜  bitrise-step-rustup git:(main) bitrise run audit-this-step                                         

  ██████╗ ██╗████████╗██████╗ ██╗███████╗███████╗
  ██╔══██╗██║╚══██╔══╝██╔══██╗██║██╔════╝██╔════╝
  ██████╔╝██║   ██║   ██████╔╝██║███████╗█████╗
  ██╔══██╗██║   ██║   ██╔══██╗██║╚════██║██╔══╝
  ██████╔╝██║   ██║   ██║  ██║██║███████║███████╗
  ╚═════╝ ╚═╝   ╚═╝   ╚═╝  ╚═╝╚═╝╚══════╝╚══════╝

  version: 1.49.3

INFO[19:01:46] bitrise runs in Secret Filtering mode 
INFO[19:01:46] bitrise runs in Secret Envs Filtering mode 
INFO[19:01:46] Running workflow: audit-this-step 

Switching to workflow: audit-this-step

INFO[19:01:46] Step uses latest version -- Updating StepLib ... 
+------------------------------------------------------------------------------+
| (0) script                                                                   |
+------------------------------------------------------------------------------+
| id: script                                                                   |
| version: 1.2.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2022-08-08T19:01:49+02:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
+ stepman audit --step-yml ./step.yml
INFO[19:01:49]  * [OK] Success audit (./step.yml) 
|                                                                              |
+---+---------------------------------------------------------------+----------+
| ✓ | script                                                        | 2.45 sec |
+---+---------------------------------------------------------------+----------+


+------------------------------------------------------------------------------+
|                               bitrise summary                                |
+---+---------------------------------------------------------------+----------+
|   | title                                                         | time (s) |
+---+---------------------------------------------------------------+----------+
| ✓ | script                                                        | 2.45 sec |
+---+---------------------------------------------------------------+----------+
| Total runtime: 2.45 sec                                                      |
+------------------------------------------------------------------------------+


Submitting anonymized usage information...
For more information visit:
https://github.com/bitrise-io/bitrise-plugins-analytics/blob/master/README.md

Bitrise build successful
```


**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/bitrise-io/bitrise-steplib?pullRequest=3572) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/bitrise-io/bitrise-steplib?pullRequest=3572) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/bitrise-io/bitrise-steplib?pullRequest=3572) it.</sub>

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    summary: Installs and uses a nightly version of the toolchain
+    title: Use Rustup Nightly
+    value_options:
+    - true
```
```suggestion
    - "true"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    title: Use Rustup Nightly
+    value_options:
+    - true
+    - false
```
```suggestion
    - "false"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    value_options:
+    - true
+    - false
+  use_rustup_nightly: false
```
```suggestion
  use_rustup_nightly: "false"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    - true
+    - false
+  use_rustup_nightly: false
+- auto_update_toolchain: false
```
```suggestion
- auto_update_toolchain: "false"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    summary: Automatically update the Rust Toolchain on every run.
+    title: Update the Rust Toolchain
+    value_options:
+    - true
```
```suggestion
    - "true"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    title: Update the Rust Toolchain
+    value_options:
+    - true
+    - false
```
```suggestion
    - "false"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    summary: Shows the version of the Toolchain components at the end of the step.
+    title: Show Toolchain versions
+    value_options:
+    - true
```
```suggestion
    - "true"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    title: Show Toolchain versions
+    value_options:
+    - true
+    - false
```
```suggestion
    - "false"
```

ofalvai commented on steps/rustup-install/0.1.0/step.yml:
```diff
@@ -0,0 +1,77 @@
+    value_options:
+    - true
+    - false
+  show_exported_envs: false
```
```suggestion
  show_exported_envs: "false"
```

ofalvai reviewed (changes requested):
Looking good @nick0602, thank you for creating this new step 🚀 

Currently the CI validation is failing because the boolean input types should be wrapped in quotes for historical reasons.

nick0602 commented:
> Looking good @nick0602, thank you for creating this new step 🚀
> 
> Currently the CI validation is failing because the boolean input types should be wrapped in quotes for historical reasons.

Apologies for that! I applied the suggestions, fixed the boolean input values in the step repo as well, and updated the SHA in this fork to reflect the tag changes (https://github.com/nick0602/bitrise-step-rustup/releases/tag/0.1.0). 

Audit run if needed:

```shell 
➜  bitrise-step-rustup git:(main) ✗ bitrise run audit-this-step

  ██████╗ ██╗████████╗██████╗ ██╗███████╗███████╗
  ██╔══██╗██║╚══██╔══╝██╔══██╗██║██╔════╝██╔════╝
  ██████╔╝██║   ██║   ██████╔╝██║███████╗█████╗
  ██╔══██╗██║   ██║   ██╔══██╗██║╚════██║██╔══╝
  ██████╔╝██║   ██║   ██║  ██║██║███████║███████╗
  ╚═════╝ ╚═╝   ╚═╝   ╚═╝  ╚═╝╚═╝╚══════╝╚══════╝

  version: 1.49.3

INFO[14:11:27] bitrise runs in Secret Filtering mode 
INFO[14:11:27] bitrise runs in Secret Envs Filtering mode 
INFO[14:11:27] Running workflow: audit-this-step 

Switching to workflow: audit-this-step

INFO[14:11:27] Step uses latest version -- Updating StepLib ... 
+------------------------------------------------------------------------------+
| (0) script                                                                   |
+------------------------------------------------------------------------------+
| id: script                                                                   |
| version: 1.2.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2022-08-09T14:11:30+02:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
+ stepman audit --step-yml ./step.yml
INFO[14:11:30]  * [OK] Success audit (./step.yml) 
|                                                                              |
+---+---------------------------------------------------------------+----------+
| ✓ | script                                                        | 2.64 sec |
+---+---------------------------------------------------------------+----------+


+------------------------------------------------------------------------------+
|                               bitrise summary                                |
+---+---------------------------------------------------------------+----------+
|   | title                                                         | time (s) |
+---+---------------------------------------------------------------+----------+
| ✓ | script                                                        | 2.64 sec |
+---+---------------------------------------------------------------+----------+
| Total runtime: 2.64 sec                                                      |
+------------------------------------------------------------------------------+


Submitting anonymized usage information...
For more information visit:
https://github.com/bitrise-io/bitrise-plugins-analytics/blob/master/README.md
```

CI check now passes but let me know if let me know if any other change is required, thanks a lot :rocket:

ofalvai reviewed (approved):
Looks good, thank you for the contribution @nick0602! 💯
